include/bluetooth/services/hids.h

```diff
@@ -167,6 +167,13 @@ struct bt_hids_rep {
  */
 typedef void (*bt_hids_notify_handler_t) (enum bt_hids_notify_evt evt);
 
+/** @brief HID notification event handler, with report identification.
+ *
+ * @param report_id Report ID defined in the HIDS Report Map.
+ * @param evt Notification event.
+ */
+typedef void (*bt_hids_notify_ext_handler_t) (uint8_t report_id, enum bt_hids_notify_evt evt);
```

Comment:
I would consider deprecating the old approach to keep API simple. Similarly as here: https://github.com/nrfconnect/sdk-nrf/pull/19692#discussion_r1900656782

Reply:
no deprecation - explained in the other PR